pyproject.toml

```diff
@@ -2,13 +2,38 @@
 requires = ['setuptools', 'wheel', 'requests']
 build-backend = 'setuptools.build_meta'
 
-[tool.black]
+[tool.ruff]
+target-version = "py38"
 line-length = 125
 
-[tool.isort]
-no_sections = true
-force_alphabetical_sort = true
-combine_as_imports = true
-profile = "black"
-skip_gitignore = true
-line_length = 125
+[tool.ruff.lint]
+extend-select = [
+  "F",
+  "E",
+  "I",
+  "UP",
+  "BLE",
+  "C4",
+  "Q",
+  "TID",
+  "PL",
+  "RUF100", # unused-noqa
+]
+extend-ignore = [
+  "PLR0904", # too-many-public-methods
+  "PLR0912", # too-many-branches
+  "PLR0913", # too-many-arguments
+  "PLR0914", # too-many-locals
+  "PLR0915", # too-many-statements
+  "PLR0916", # too-many-boolean-expressions
+  "PLR2004", # magic-value-comparison
+  "UP006", # non-pep585-annotation
+  "UP007", # non-pep604-annotation
```

Comment:
question (non-blocking): Why not address these?

Reply:
I prefer to fix those in a separate PR as some files use postponed evaluation of annotations while others don't, so one needs to be careful. Note that the minimum supported python version is 3.8, which predates PEP 585 and PEP 604.